### PR TITLE
rados: Set snappy as default value in ms_osd_compression_algorithm

### PR DESCRIPTION
**Pull request by shreyanshjain7174**
rados: Set snappy as a default value in ms_osd_compression_algorithm instead of list of algorithms.

 The OSDs choose the preferred algorithm based on mutual understanding between OSDs. The proposed solution to this problem was to set "snappy" as default algorithm and provide flexibility to user to change the algorithm of their choice. The long description for the global file is also updated.
    
**Fixes: https://tracker.ceph.com/issues/58410**
Signed-off-by: shreyanshjain7174 <ssanchet@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [x] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [x] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/common/options/global.yaml.in [hunk 1]
```diff
@@ -1018,8 +1018,9 @@ options:
   type: str
   level: advanced
   desc: Compression algorithm to use in Messenger when communicating with OSD
-  long_desc: Compression algorithm for connections with OSD in order of preference
-  default: snappy zlib zstd lz4
+  long_desc: Compression algorithm has been set to snappy as the default 
+    algorithm. However, there is flexibility to set the algorithm manually.
```
Comment (rzarzynski) [inline src/common/options/global.yaml.in:1022]: The commit shouldn't remove the original message that was explaining the purpose of the configurable. I would just add that, although the default value is singular, a list (like `snappy zlib zstd` etc.) is acceptable as well.

**Review by rzarzynski — CHANGES_REQUESTED** [review]

File: src/common/options/global.yaml.in (hunk 1 above)
Comment (shreyanshjain7174) [reply src/common/options/global.yaml.in]: Done!

File: src/common/options/global.yaml.in [hunk 2]
```diff
@@ -1018,8 +1018,10 @@ options:
   type: str
   level: advanced
   desc: Compression algorithm to use in Messenger when communicating with OSD
-  long_desc: Compression algorithm for connections with OSD in order of preference
-  default: snappy zlib zstd lz4
+  long_desc: Compression algorithm for connections with OSD in order of preference 
+    Although, the default value is set to snappy,however a list
```
Comment (rzarzynski) [inline src/common/options/global.yaml.in:1022]:
```suggestion
    Although the default value is set to snappy, a list
```

**Review by rzarzynski — APPROVED** [review]

**Comment by ljflores** [comment]
Rados suite review: http://pulpito.front.sepia.ceph.com/?branch=wip-yuri10-testing-2023-02-15-1245

Failures, unrelated:
    1. https://tracker.ceph.com/issues/58585
    2. https://tracker.ceph.com/issues/58475
    3. https://tracker.ceph.com/issues/58797 -- new tracker; seen in main baseline, therefore unrelated to trackers in this batch

Details:
    1. rook: failed to pull kubelet image - Ceph - Orchestrator
    2. test_dashboard_e2e.sh: Conflicting peer dependency: postcss@8.4.21 - Ceph - Mgr - Dashboard
    3. scrub/osd-scrub-dump.sh: TEST_recover_unexpected fails from "ERROR: Unexpectedly low amount of scrub reservations seen during test" - Ceph - RADOS

**Pull request merged**